### ISSUE TITLE
Update to Gradle 8.0 and Android 34 requires build system update

### ISSUE THREAD

**Issue by codeanticode** [processing/processing-android#767]
Right now the entire build system is stuck at Gradle 7.2 and Android 33 because anything higher would require moving to Gradle 8. which is not backwards-compatible.

**Comment by codeanticode** [comment]
So this needs to be fixed asap :-)

**Comment by SableRaf** [comment]
Thanks for the heads up @sandrobenigno 🙏

@codeanticode: what amount of work would be needed for this update?

cc @ranaaditya @stefterv

**Comment by codeanticode** [comment]
It may take a substantial amount of work because there is quite a bit of custom Gradle code that needs to be updated to be compatible with 8.0.

**Comment by sandrobenigno** [comment]
(FYI) From now on, Google will no longer accept apps created using the current Processing for Android mode.
I asked for an extension for my app, but now they no longer allow updates or even closed testing.

**Comment by codeanticode** [comment]
@sandrobenigno I'm sorry that there is no solution currently for this issue, I will do my best to find a time sometime soon (although I don't think I will be able to work on it until next month or so). For the time being, a workaround would be to export your sketch as Gradle project, so you continue the development from Android Studio. You should be able to fully utilize the Processing API, while upgrading the latest version of Gradle, etc. to pass through Google's latest review requirements.

**Comment by codeanticode** [comment]
@Stefterv yes that would be a great resource, thank you. The processing android book has an appendix precisely on that that you could use as a starting point.

**Comment by JakubValtar** [comment]
@SableRaf Thanks for the ping, sadly I'm not up to date with the state of the Android SDK and don't have the bandwidth right now to look into it. I hope somebody else will be able to pick this up!

**Comment by SableRaf** [comment]
I made an (uneducated) attempt to update to Gradle 8.0 and Android 34 together, but I ran into a few issues. One tricky problem seems to be that GoogleVR is deprecated, which creates compatibility issues with Gradle 8 and Android 34. I know @p4puniya put a lot of effort into the Cardboard SDK migration, but they hit some roadblocks and would take more work to finish so we don't have an obvious path forward right now.

@codeanticode: Would it make sense to temporarily drop VR support if that helps people keep shipping apps to the Play Store? Maybe @phanirithvij or @ranaaditya have some thoughts too?

PS: There’s an untested draft PR made with @morisil's custom AI agent (using Claude 3.5) at https://github.com/processing/processing-android/pull/773

**Comment by morisil** [comment]
I believe that due the the knowledge cut off of the LLM the SDK version is not the latest one, but 33. At least claudine was able to make the code pass gradle build, after I manually removed one module from the build.

**Comment by p4puniya** [comment]
Hi all, so something to note about the new Cardboard SDK is that It can't be integrated with the older versions of processing as I've mentioned this earlier as well, since it's a library that's build on C++ and Processing is a Java Multi-module library. What can be done is create a new library that works entirely on the newer versions of Cardboard SDK and Gradle and build custom tutorials for the same. 
Ps. This is as far as Google VR & Cardboard SDK is concerned. 

About the issue with Gradle 8.0 compatibility, when I was working on P-Android, I had to download and manually import multiple versions of different libraries in order to get the Android Library working. One thing that I suggest here is, that instead of trying to build backward compatibility with the already existing apps, we can build a similar library with all the updated SDKs and dependencies. (Like a fresh new start for the Android Library)
This will be a time taking process and we will have educate the audience regarding the new changes, but I do believe that this is something worth investing in. This will also allow ease of access for the new generation of developers to start working processing android, as currently, just setting up the work environment with all the dependencies and libraries breaking seems like a really big roadblock for anyone starting afresh. 
I know that this may seem like too much work, but If you are considering the bigger picture, this can be a much needed upgrade for the Android Library. 
What are your thoughts? @SableRaf @codeanticode @ranaaditya

**Comment by TolyaDenisenko** [comment]
Hi, VR examples continue to be non-working. Are there any solutions?

**Comment by SableRaf** [comment]
Hi @TolyaDenisenko. You can try using older versions of Android and Processing for Android. Besides that, I'm afraid there isn't an easy workaround yet.

_Note: The Gradle migration of Processing [has been merged](https://github.com/processing/processing4/pull/888) and will be released as Beta soon. It won't fix the issue above by itself, but it is a first step towards better integration with Android in Processing core._

**Comment by TolyaDenisenko** [comment]
> Привет[@TolyaDenisenko](https://github.com/TolyaDenisenko). Вы можете попробовать использовать более старые версии Android и Processing for Android. Кроме того, боюсь, что пока нет простого обходного пути.
> 
> _Примечание: Gradle-миграция Processing [была объединена](https://github.com/processing/processing4/pull/888) и скоро будет выпущена как Beta. Она сама по себе не исправит указанную выше проблему, но это первый шаг к лучшей интеграции с Android в ядре Processing._

That is, I can take an older smartphone and try to run it. Tell me what is the latest supported version of Android?
